src/cli/mod.rs

```diff
@@ -0,0 +1,125 @@
+pub mod mirror;
+pub mod scan;
+
+use crate::{self as stellar_archivist, mirror_operation, scan_operation};
+use clap::{Parser, Subcommand};
+use std::ffi::OsString;
+use std::io;
+use thiserror::Error;
+use tracing::Level;
+use tracing_subscriber::FmtSubscriber;
+
+/// CLI errors
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Clap(#[from] clap::Error),
+
+    #[error(transparent)]
+    Io(#[from] io::Error),
+
+    #[error(transparent)]
+    ScanOperation(#[from] scan_operation::Error),
+
+    #[error(transparent)]
+    MirrorOperation(#[from] mirror_operation::Error),
+
+    #[error("{0}")]
+    Other(String),
+}
+
+impl From<stellar_archivist::Error> for Error {
+    fn from(err: stellar_archivist::Error) -> Self {
+        match err {
+            stellar_archivist::Error::Io(e) => Error::Io(e),
+            stellar_archivist::Error::ScanOperation(e) => Error::ScanOperation(e),
+            stellar_archivist::Error::MirrorOperation(e) => Error::MirrorOperation(e),
+            stellar_archivist::Error::Other(s) => Error::Other(s),
+        }
+    }
+}
+
+#[derive(Parser)]
+#[command(name = "stellar-archivist")]
+#[command(about = "Stellar History Archive tools and utilities", long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+
+    /// Number of files to process concurrently
+    #[arg(short, long, global = true, default_value_t = 32)]
+    concurrency: usize,
+
+    /// Skip optional SCP files
+    #[arg(long, global = true)]
+    skip_optional: bool,
+
+    /// Enable debug logging
+    #[arg(long, global = true)]
+    debug: bool,
+
+    /// Enable trace logging
+    #[arg(long, global = true)]
+    trace: bool,
+
+    /// Maximum number of HTTP retry attempts
+    #[arg(long, global = true, default_value_t = 3)]
+    max_retries: u32,
+
+    /// Initial backoff in milliseconds for HTTP retries
+    #[arg(long, global = true, default_value_t = 100)]
+    initial_backoff_ms: u64,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Mirror files from source archive to destination
+    Mirror(mirror::MirrorCmd),
```

Comment:
Fyi, up to you, but the comment on the Mirror enum here, can also be placed on the MirrorCmd, which has the advantage that all the help docs for the MirrorCmd, both the description and the options, will appear in the same place. Mentioning because this wasn't obvious to me as possible and I only found this out late in the stellar-cli's dev.